CMakeLists.txt

```diff
@@ -28,7 +28,16 @@ if(BINARY_RELEASE)
   set(DEBUG OFF)
   set(DIST ON)
   set(FASTER OFF)
-  set(CMAKE_INTERPROCEDURAL_OPTIMIZATION ON)
+
+  # Use LTO on all compilers where it is supported.
+  include(CheckIPOSupported)
+  check_ipo_supported(RESULT result)
+  if(result)
+    set(CMAKE_INTERPROCEDURAL_OPTIMIZATION ON)
+    message(STATUS "LTO enabled")
+  else()
+    message(WARNING "LTO not supported by this compiler and/or CMake version")

```

Comment:
Cant decide if this is needed or not.

Reply:
This should be rare nowadays but I guess it's useful to get this warning if you're building a release and somehow have a very old compiler

Reply:
Yeah guess it works as a "things could be better" notice. But then do we need the "LTO enabled" message :D

Reply:
Probably not, feel free to remove it :)